### PR TITLE
solved

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -8,7 +14,26 @@
...
+    private final String emptyString = "";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        final Path path = Paths.get(fileName);
+        try {
+            if (Files.exists(path)
+                    && Files.isReadable(path)
+                    && Files.size(path) > 0) {
+                return Arrays.stream(Files.readString(path).split(space))
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:28]: you are splitting by " " here
do words separated by "\n" get split?

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -8,7 +14,26 @@
...
  */
 public class FileWork {
+    private final String regex = "\\W";
+    private final String letter = "w";
+    private final String space = " ";
+    private final String emptyString = "";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        final Path path = Paths.get(fileName);
```
[the remark this answers is not in the document]
Comment (ilya449) [reply src/main/java/core/basesyntax/FileWork.java:23]: because it doesn't change

File: src/main/java/core/basesyntax/FileWork.java (hunk 1 above)
Comment (ilya449) [reply src/main/java/core/basesyntax/FileWork.java:28]: yes

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -8,7 +14,26 @@
  * Результат: web wide width world</p>
  */
 public class FileWork {
+    private final String regex = "\\W";
+    private final String letter = "w";
+    private final String space = " ";
+    private final String emptyString = "";
```
[the remark this answers is not in the document]
Comment (ilya449) [reply src/main/java/core/basesyntax/FileWork.java:20]: when I named them in UPPER_CASE I had problems with check style.
but now I've made them static and everything is OK, thank you

File: src/main/java/core/basesyntax/FileWork.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/FileWork.java:23]: I think `final` is redundant here
it is used when we explicitly indicate that we want a variable to never point at any other value

File: src/main/java/core/basesyntax/FileWork.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/FileWork.java]: unfortunately no
`Files.readString(path)` - here you get one big line
and line breaks look the following  - `\n'

when you do `.split(" ") then, the words separated with `\n' remain together

File: src/main/java/core/basesyntax/FileWork.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ilya449) [reply src/main/java/core/basesyntax/FileWork.java]: Ok, I've changed it

File: src/main/java/core/basesyntax/FileWork.java (hunk 1 above)
Comment (ilya449) [reply src/main/java/core/basesyntax/FileWork.java]: I've just tried to split string line which contains several ```\n``` and ```" "```  by ```.split(" ")``` and it was splitted everywhere...

Comment (ilya449) [reply src/main/java/core/basesyntax/FileWork.java]: Do you mean that I need to use ```\\s+``` instead ```" "```? I've found it and changed

**Pull request closed without merging**